core/provisioning-java/src/main/java/org/apache/syncope/core/provisioning/java/data/ClientAppDataBinderImpl.java

```diff
@@ -226,96 +172,59 @@ private void doUpdate(final OIDCRP clientApp, final OIDCRPTO clientAppTO) {
-                } else {
-                    sce.getElements().add("Expected " + AccessPolicy.class.getSimpleName()
-                            + ", found " + policy.getClass().getSimpleName());
-                }
-                throw sce;
-            }
-        }
-
-        if (clientAppTO.getAttrReleasePolicy() == null) {
-            clientApp.setAttrReleasePolicy(null);
-        } else {
-            Policy policy = policyDAO.find(clientAppTO.getAttrReleasePolicy());
-            if (policy instanceof AttrReleasePolicy) {
-                clientApp.setAttrReleasePolicy((AttrReleasePolicy) policy);
-            } else {
-                SyncopeClientException sce = SyncopeClientException.build(ClientExceptionType.InvalidPolicy);
-                if (policy == null) {
-                    sce.getElements().add("Policy " + clientAppTO.getAttrReleasePolicy() + " not found");
-                } else {
-                    sce.getElements().add("Expected " + AttrReleasePolicy.class.getSimpleName()
-                            + ", found " + policy.getClass().getSimpleName());
-                }
-                throw sce;
-            }
-        }
-
         clientApp.setLogoutUri(clientAppTO.getLogoutUri());
     }
 
     private static OIDCRPTO getClientAppTO(final OIDCRP clientApp) {
         OIDCRPTO clientAppTO = new OIDCRPTO();
+        updateCommonClientAppTO(clientApp, clientAppTO);
 
-        clientAppTO.setName(clientApp.getName());
-        clientAppTO.setKey(clientApp.getKey());
-        clientAppTO.setDescription(clientApp.getDescription());
-        clientAppTO.setClientAppId(clientApp.getClientAppId());
         clientAppTO.setClientId(clientApp.getClientId());
         clientAppTO.setClientSecret(clientApp.getClientSecret());
         clientAppTO.setSignIdToken(clientApp.isSignIdToken());
         clientAppTO.setSubjectType(clientApp.getSubjectType());
         clientAppTO.getRedirectUris().addAll(clientApp.getRedirectUris());
         clientAppTO.getSupportedGrantTypes().addAll(clientApp.getSupportedGrantTypes());
         clientAppTO.getSupportedResponseTypes().addAll(clientApp.getSupportedResponseTypes());
-
-        clientAppTO.setAuthPolicy(clientApp.getAuthPolicy() == null
-                ? null : clientApp.getAuthPolicy().getKey());
-        clientAppTO.setAccessPolicy(clientApp.getAccessPolicy() == null
-                ? null : clientApp.getAccessPolicy().getKey());
-        clientAppTO.setAttrReleasePolicy(clientApp.getAttrReleasePolicy() == null
-                ? null : clientApp.getAttrReleasePolicy().getKey());
-
         clientAppTO.setLogoutUri(clientApp.getLogoutUri());
 
         return clientAppTO;
     }
 
     private void doUpdate(final CASSP clientApp, final CASSPTO clientAppTO) {
+        doUpdateCommon(clientApp, clientAppTO);
+
+        clientApp.setServiceId(clientAppTO.getServiceId());
+    }
+
+    private static CASSPTO getClientAppTO(final CASSP clientApp) {
```

Comment:
@mmoayyed could you please have a look at this?

Reply:
I mean this: https://github.com/apache/syncope/pull/223/checks?check_run_id=1314109065